sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/persons/PersonDirectorySteps.java

```diff
@@ -121,10 +245,53 @@ public PersonDirectorySteps(
           webDriverHelpers.isElementVisibleWithTimeout(UUID_INPUT, 20);
         });
 
-    // TODO Michal, due to specific logic to filter only for this situation, create a method with a
-    // suggestive name, where you apply filters based on generated POJO data.
-    // We can't create multiple methods to reuse them in this case, but please keep in mind to
-    // create generic locators
-    // The above method will need to be finished by you (the switch to cover all search criteria)
+    Then(
+        "I search after last created person from API by factor {string}",
+        (String searchCriteria) -> {
+          String searchText = "";
+          String personUUID = apiState.getLastCreatedPerson().getUuid();
+          switch (searchCriteria) {
+            case "uuid":
+              searchText = dataOperations.getPartialUuidFromAssociatedLink((personUUID));
+              break;
+            case "full name":
+              searchText =
+                  apiState.getLastCreatedPerson().getLastName()
+                      + " "
+                      + apiState.getLastCreatedPerson().getFirstName();
+              break;
+            case "phone number":
+              searchText = apiState.getLastCreatedPerson().getPhone();
+              break;
+            case "email":
+              searchText = apiState.getLastCreatedPerson().getEmailAddress();
+              break;
+          }
+
+          webDriverHelpers.fillInWebElement(MULTIPLE_OPTIONS_SEARCH_INPUT, searchText);
+        });
+
+    Then(
+        "I change {string} information data field for Person",
```

Comment:
this steps confused me a bit, I don't understand exactly what it does because the uuid cannot be changed